packages/matter.js/src/CommissioningController.ts

```diff
@@ -132,10 +133,33 @@ export class CommissioningController extends MatterNode {
                 identifierData = {};
             }
 
-            this.nodeId = await this.controllerInstance.commission(identifierData, this.passcode, 120, this.serverAddress);
+            let bleEnabled = false;
+            try {
+                bleEnabled = !!Ble.get();
+            } catch (e) {
+                logger.warn(`Ble not enabled: ${e}`);
+            }
+
+            // TODO: Make the process more parallel and favor MDNS over BLE, but for testing right now lets do that way
+            let nodeId: NodeId | undefined;
+            if (bleEnabled) {
```

Comment:
I'm also not happy with it.
What we would need is a further adjustment, thats why i added the TODO for now - also because it is a bit unclear how we want controller to behave in future really.

In fact it should be "discover in parallel" and who ever discovers first that one wins and continues ... buut the more unspecific you search (e.g. by just discover all commissionable and try passcodes by doing a Pase session).
So thats a bigger change I planned for later - also becausde right now BLE is more a showcase and if the devcie in question is already in the network then most likely mdns will always win :)

I enhanced the TODO a bit for now, but I would leave it that way. The readme states that when -ble is used BLE is tried first with 15s timeout before mdns is tried

Reply:
Cool, I thought maybe comment just meant during development